Build check not passing
```
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c: In function ‘main’:
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c:53:44: error: invalid type argument of ‘->’ (have ‘card_t’)
   53 |                         players[i].cards[j]->suit=c/4;
      |                                            ^~
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c:54:44: error: invalid type argument of ‘->’ (have ‘card_t’)
   54 |                         players[i].cards[j]->value=(c%13)+2;
      |                                            ^~
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c:58:31: error: ‘j’ undeclared (first use in this function)
   58 |             for (int i=0; i<5;j++){
      |                               ^
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c:58:31: note: each undeclared identifier is reported only once for each function it appears in
/home/runner/work/projekt-zespol-nr-14/projekt-zespol-nr-14/src/strange_poker.c:61:17: error: ‘tableCards’ undeclared (first use in this function)
   61 |                 tableCards[i]->suit=c/4;
      |                 ^~~~~~~~~~
gmake[2]: *** [CMakeFiles/StrangePoker.dir/build.make:76: CMakeFiles/StrangePoker.dir/src/strange_poker.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:100: CMakeFiles/StrangePoker.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

--- a/src/strange_poker.c
+++ b/src/strange_poker.c
@@ -48,22 +48,22 @@ int main(int argc, char *argv[])
             //dealing cards
             
             for(int i=0; i<52;i++){
-                *(tab+i)=0;
+                *(card+i)=0;
             }
             for (uint32_t i = 0; i < playerCnt; i++){
                 for (int j=0; j<5;j++){
                     int c=rand() % 52;
                        while(*(card+c)!=0) c=rand() % 52;
-                        players[i]->cards[j]->suit=c/4;
-                        players[i]->cards[j]->value=(c%13)+2;
+                        players[i].cards[j].suit=c/4;
+                        players[i].cards[j].value=(c%13)+2;
                         *(card+i)=1;
                 }
             }
             for (int i=0; i<5;i++){
                 int c=rand() % 52;
                 while(*(card+c)!=0) c=rand() % 52;
-                ctx.tableCards[i]->suit=c/4;
-                ctx.tableCards[i]->value=(c%13)+2;
+                ctx.tableCards[i].suit=c/4;
+                ctx.tableCards[i].value=(c%13)+2;
                 *(card+i)=1;
             }
             

--- a/src/strange_poker.c
+++ b/src/strange_poker.c
@@ -4,7 +4,6 @@
 #include <stdlib.h>
 #include <unistd.h>
 
-int cards[52];
 
 int main(int argc, char *argv[])
 {
@@ -18,6 +17,7 @@ int main(int argc, char *argv[])
         return -1;
     }
     srand((unsigned int) time(NULL));
+    int *card=malloc(sizeof(int)*52);
     
     while( 1 ) 
     {
@@ -46,26 +46,32 @@ int main(int argc, char *argv[])
             }
             
             //dealing cards
+            
+            for(int i=0; i<52;i++){
+                *(tab+i)=0;
+            }
             for (uint32_t i = 0; i < playerCnt; i++){
                 for (int j=0; j<5;j++){
                     int c=rand() % 52;
-                       while(cards[c]!=0) c=rand() % 52;
-                        players[i].cards[j]->suit=c/4;
-                        players[i].cards[j]->value=(c%13)+2;
-                        cards[c]=1;
+                       while(*(card+c)!=0) c=rand() % 52;
+                        players[i]->cards[j]->suit=c/4;
+                        players[i]->cards[j]->value=(c%13)+2;
+                        *(card+i)=1;
                 }
             }
-            for (int i=0; i<5;j++){
+            for (int i=0; i<5;i++){
                 int c=rand() % 52;
-                while(cards[c]!=0) c=rand() % 52;
-                tableCards[i]->suit=c/4;
-                tableCards[i]->value=(c%13)+2;
-                cards[c]=1;
+                while(*(card+c)!=0) c=rand() % 52;
+                ctx.tableCards[i]->suit=c/4;
+                ctx.tableCards[i]->value=(c%13)+2;
+                *(card+i)=1;
             }
+            
 
             // ?????
         }
     }
+    free(card);
 
     ui->destroy( ui->data );
 }

--- a/src/strange_poker.c
+++ b/src/strange_poker.c
@@ -4,6 +4,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 
+char cards[52];
 
 int main(int argc, char *argv[])
 {
@@ -17,7 +18,6 @@ int main(int argc, char *argv[])
         return -1;
     }
     srand((unsigned int) time(NULL));
-    int *card=malloc(sizeof(int)*52);
     
     while( 1 ) 
     {
@@ -48,23 +48,23 @@ int main(int argc, char *argv[])
             //dealing cards
             
             for(int i=0; i<52;i++){
-                *(card+i)=0;
+                cards[i]='0';
             }
             for (uint32_t i = 0; i < playerCnt; i++){
                 for (int j=0; j<5;j++){
                     int c=rand() % 52;
-                       while(*(card+c)!=0) c=rand() % 52;
+                       while(cards[c]!='0') c=rand() % 52;
                         players[i].cards[j].suit=c/4;
                         players[i].cards[j].value=(c%13)+2;
-                        *(card+i)=1;
+                        cards[c]='1';
                 }
             }
             for (int i=0; i<5;i++){
                 int c=rand() % 52;
-                while(*(card+c)!=0) c=rand() % 52;
+                while(cards[c]!='0') c=rand() % 52;
                 ctx.tableCards[i].suit=c/4;
                 ctx.tableCards[i].value=(c%13)+2;
-                *(card+i)=1;
+                cards[c]='1';
             }
             
 

